client/network/src/protocol/notifications/behaviour.rs

```diff
@@ -911,48 +1003,55 @@ impl Notifications {
 		}
 	}
 
+	/// Function that is called when the protocol wants us to reject an incoming peer.
+	fn protocol_report_reject(&mut self, index: IncomingIndex) {
+		if let Some((set_id, peer_id)) = self.peerset_report_reject(index) {
+			self.protocol_controller_handles[usize::from(set_id)].dropped(peer_id)
+		}
```

Comment:
I'm not entirely sure why this naming is confusing so could you perhaps elaborate?

`peerset_report_reject()` rejects the connection (like it did before) without informing the protocol about the inbound connection (there is no point). `protocol_report_reject()` gets called if protocol rejects the connection and it must perform the same steps as `peerset_report_reject()` but additionally remove the peer from `ProtocolController` since it now thinks the peer is connected.

Reply:
It's 100% subjective, so feel free to ignore. I just wouldn't call something with a name `peerset_report_reject` from something actually called not by peerset but a protocol. So, I would rename `peerset_report_reject` to just `report_reject`, call it as is from `peerset_report_reject`, and call this `report_reject` from `protocol_report_reject` to make it clear that this time it is not reported by peerset. Not that important at all, so feel free to ignore.

Reply:
Fair point, renamed